assets/src/eejs/index.js

```diff
@@ -21,9 +21,13 @@ export const i18n = wpI18n;
 
 /**
  * use this for eejs exceptions
- * Usage: throw new eejs.exception('some message')
+ * Usage: throw new eejs.Exception('some message')
  * @param {string} msg
  */
-export const exception = function( msg ) {
+export const Exception = function( msg ) {
 	this.message = msg;
+	this.name = 'Exception';
+	this.stack = ( new Error() ).stack;
 };
+
+Exception.prototype = new Error;
```

Comment:
I tweaked the current generic EE exception error handler after doing some more reading up on it.  This makes it inherit more properties of the native `Error` class in javascript.  We can use this pattern for any custom Exceptions we want to create for client side error handling (such as comparables to what we have PHP side).  I think its valuable to keep all these exceptions exposed on the `eejs` global (so `eejs.Exception` for instance), so our add-ons can take advantage of them as well.  If start increasing the number of custom exception classes we can break them out into their own file and just export them from here.

Reply:
agreed. sounds great